vcert/connection_cloud.py

```diff
@@ -477,6 +478,43 @@ def revoke_cert(self, request):
         # not supported in Venafi Cloud
         raise NotImplementedError
 
+    def retire_cert(self, request):
+        cert_id = None
+        if not request.id and not request.thumbprint:
+            log.error("id or thumbprint must be specified for retiring certificate")
+            raise ClientBadData
+
+        if request.thumbprint:
+            response = self.search_by_thumbprint(request.thumbprint)
+            cert_ids = response.certificateIds
+            if len(cert_ids) > 1:
+                log.error(f"multiple certificates matching thumbprint found")
+                raise VenafiError
+            cert_id = cert_ids[0]
+
+        if request.id:
```

Comment:
You should do this comparison first, if request.id is present, then no need to lookup the id using the thumbprint.

Reply:
Also to add, the `request.thumbprint` validation should be an `elif` otherwise you will still try to look from certificate ID with thumprint regardless of already having the certificate ID.